src/InstallerHosts/Elastic.InstallerHosts/Elasticsearch/Tasks/Rollback/RollbackDirectoriesTask.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.IO;
+using System.IO.Abstractions;
+using Elastic.Configuration.FileBased.Yaml;
+using Elastic.Installer.Domain.Configuration.Wix.Session;
+using Elastic.Installer.Domain.Model.Elasticsearch;
+using Elastic.InstallerHosts.Elasticsearch.Tasks.Uninstall;
+
+namespace Elastic.InstallerHosts.Elasticsearch.Tasks.Rollback
+{
+	public class RollbackDirectoriesTask : DeleteDirectoriesTask
+	{
+		public RollbackDirectoriesTask(string[] args, ISession session)
+			: base(args, session) { }
+
+		public RollbackDirectoriesTask(ElasticsearchInstallationModel model, ISession session, IFileSystem fileSystem)
+			: base(model, session, fileSystem) { }
+
+		protected override bool ExecuteTask()
+		{
+			if (this.InstallationModel.NoticeModel.ExistingVersionInstalled)
+			{
+				RestoreConfigDirectory();
+				RestorePluginsDirectory();
+				return true;
+			}
+
+			var fs = this.FileSystem;
+
+			if (this.Session.IsInstalling)
```

Comment:
decrease nesting either by inverting the `if` or creating a method of the body

Reply:
Add a comment that this section will only be hit during a rollback of a fresh installation.

Reply:
> decrease nesting either by inverting the if or creating a method of the body

combined the `if`s